### PR TITLE
feat: stage the template once created

### LINKED ISSUE #1026: feat: Stage the template once created
### Is your feature request related to a problem? Please describe.

You run `package-manager-of-choice create t3-app@latest`. You work on making a quick app. You go to commit your changes, and boom you see a lot of extra files you never touched. These were from the template, they don't matter for your commit. Ideally, you would've committed the template to have a clean working tree, but it's too late now because you forgot.

### Describe the solution you'd like to see

In order of my preference, based on what others have said:

#### A.
* Stage the template

This gives the option to:
* Commit with custom message
* Commit just the template even if you made edits, since staging stores a snapshot
* Add everything (without even needing to unstage) if you want to have template + changes as your first commit

#### B.
* Prompt if user wants an initial commit (y/N) (default no)
* If yes, prompt the user for the commit message, with "Initial commit from Create T3 App"/"Initial commit from ct3a"/Initial commit from create-t3-app" (or similar) as the default entry that disappears (like the current input for directory name)

#### C.
* Prompt the user for the initial commit message (like A.2), but have <kbd>Space</kbd><kbd>Enter</kbd> or <kbd>Esc</kbd> cancel the initial commit (don't know if <kbd>Esc</kbd> is possible with inquirer)

#### D.
* Have a select dropdown with `no commit`, `"Initial commit from Create T3 App"` (or similar), or `custom commit message` as options
* If custom is selected, have a prompt like A.2
And of course, for all these options, relevant command line arguments.

#### Just don't forget

😐

#### Deal with it

🙄

#### Always commit the template

A lot of people on our Discord don't want the template committed or with a forced name, so doing this by default would require them to either deal with it, reinitialize git, or amend the commit.

### Additional information

|       Tool       |                                                       Git                                                       |
| :--------------: | :-------------------------------------------------------------------------------------------------------------: |
| create-next-app  |    Automatically creates a git repo, commits template with the message "Initial commit from Create Next App"    |
| create-react-app | Automatically creates a git repo, commits template with the message "Initialize project using Create React App" |
| create-astro-app |                                           Does not create a git repo                                            |
|   create-vite    |                                           Does not create a git repo                                            |

Discussed on Discord from [1057485366757687377](https://discord.com/channels/966627436387266600/988912020558602331/1057485366757687377) to [1057528318796234762](https://discord.com/channels/966627436387266600/988912020558602331/1057528318796234762)

### PR DESCRIPTION
**Pull request by complexengine**
Closes #1026 

## ✅ Checklist

- [x] I have followed every step in the [contributing guide](https://github.com/t3-oss/create-t3-app/blob/main/CONTRIBUTING.md) (updated 2022-10-06).
- [x] The PR title follows the convention we established [conventional-commit](https://www.conventionalcommits.org/en/v1.0.0/)
- [x] I performed a functional test on my final commit

---

## Changelog

The CLI now stages the files on project initialization.

---

### CODE REVIEW

File: cli/src/helpers/git.ts [hunk 1]
```diff
@@ -120,6 +120,7 @@ export const initializeGit = async (projectDir: string) => {
         cwd: projectDir,
       });
     }
+    await execa("git", ["add", "."], { cwd: projectDir });
     spinner.succeed(
       `${chalk.green("Successfully initialized")} ${chalk.green.bold("git")}\n`,
```
Comment (juliusmarminge) [inline cli/src/helpers/git.ts:125]: maybe change this log to `initialised and staged` or something to let users know we did that?

Comment (complexengine) [reply cli/src/helpers/git.ts]: Sure, updated commit

**Comment by Samathingamajig** [comment]
I think this should have a [changeset](https://github.com/changesets/changesets/blob/main/docs/adding-a-changeset.md) since this is significant change to the output.

**Comment by complexengine** [comment]
Added a minor changeset.

**Review by c-ehrlich — CHANGES_REQUESTED** [review]
Thanks for the PR, I think this will be an improvement over what we are currently doing :)

It currently doesn't stage two changes to `package.json` - changing the name from "template" to the name the user selected, and adding the `ct3aMetadata` object. Getting these to stage also might require changing the order in which the CLI does stuff, but I think git being the very last thing that happens makes sense overall.

image

edit: just realized I forgot to test this with installing packages through the CLI, but the git commit should happen after that also

also dont worry about autorelease failing, thats an issue on our side

**Comment by complexengine** [comment]
Fixed @c-ehrlich

**Review by c-ehrlich — APPROVED** [review]
Works great now including package.json and lockfile. Thanks for the contribution!

**Pull request merged**